### PR TITLE
Creating get method for user profile, post method for edit profile/change password 

### PR DESCRIPTION
**Pull request by avolchivschi**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -30,7 +30,7 @@ export default {
   computed: {
     isAuth() {
       const { name } = this.$route;
-      return name === 'login' || name === 'registration';
+      return name === 'login' || name === 'registration' || name === 'error';
```
Comment (nlaptedulce) [inline src/App.vue]: rename not-found or 404

File: src/components/auth/VError.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<template>
```
Comment (nlaptedulce) [inline src/components/auth/VError.vue]: rename please this view

File: src/components/user/ChangePassword.vue [hunk 3]
```diff
@@ -1,39 +1,118 @@
...
           class="form-control"
+          :class="{ 'is-invalid': errors.has('old_password') }"
         />
+
+        <span v-if="errors.has('old_password')" class="invalid-feedback">
+          {{ 'Incorrect old password' }}
+        </span>
       </b-form-group>
 
+
```
Comment (nlaptedulce) [inline src/components/user/ChangePassword.vue]: to much lines

File: src/components/user/ChangePassword.vue [hunk 4]
```diff
@@ -1,39 +1,118 @@
...
+        old_password: '',
+        password: '',
+        confirm_password: '',
+      },
+    }),
+    methods: {
+      change() {
+        UserApi.changePassword(this.form)
+          .then(response => {
+            console.log(response);
```
Comment (nlaptedulce) [inline src/components/user/ChangePassword.vue]: remove this console.log

File: src/components/user/UserDetails.vue [hunk 5]
```diff
@@ -1,66 +1,178 @@
...
+        id="input-group-1"
         class="col-md-6"
-        label="First Name"
+        label="Username"
+        label-for="username"
       >
         <b-form-input
+          id="username"
+          v-model="form.username"
+          v-validate.continues="'alpha_dash|required|min:5|max:20'"
```
Comment (nlaptedulce) [inline src/components/user/UserDetails.vue]: are you sure here at least 5?

File: src/services/userDetailsApi.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import Axios from 'axios';
+const API_USER= 'http://api.pentabee.local/api/user/';
+import axios from 'axios';
+
+export default {
+  mounted(){
```
Comment (nlaptedulce) [inline src/services/userDetailsApi.js]: rename this method

File: src/components/user/UserDetails.vue [hunk 7]
```diff
@@ -1,66 +1,178 @@
...
   </div>
 </template>
+
+<script>
+  import UserApi from '@/services/userDetailsApi';
+
+  export default {
+    data: () => ({
+        form: {
+          id: 52 ,
```
Comment (nlaptedulce) [inline src/components/user/UserDetails.vue]: you need get user id from token, and send in request

File: src/components/user/UserDetails.vue [hunk 8]
```diff
@@ -1,66 +1,178 @@
...
+        console.log(response);
+        this.form = response.data;
+      }).catch(error => {
+        console.log(error)
+      })
+    },
+    methods: {
+      edit() {
+        UserApi.editUser(this.form)
+        // UserApi.editUser(this.form.id)
```
Comment (nlaptedulce) [inline src/components/user/UserDetails.vue]: remove this comment

File: src/components/user/UserDetails.vue [hunk 9]
```diff
@@ -1,66 +1,178 @@
...
+          .then(response => {
+            console.log(response);
```
Comment (nlaptedulce) [inline src/components/user/UserDetails.vue]: remove this console.log

File: src/components/user/VProfile.vue [hunk 10]
```diff
@@ -1,53 +1,60 @@
...
+      </b-card-group>
     </b-container>
   </div>
 </template>
 
 <script>
   import UserDetails from './UserDetails';
   import ChangePassword from './ChangePassword';
+  import UserApi from '@/services/userDetailsApi';
+
```
Comment (nlaptedulce) [inline src/components/user/VProfile.vue]: remove line

File: src/services/userDetailsApi.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    const token = localStorage.getItem('token');
+    console.log(token);
+    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+  },
+  getUserInfo(id) {
+    return Axios.get(API_USER + id);
+  },
+  editUser(data) {
+    return Axios.post(API_USER + 52 + '/edit', data);
```
Comment (nlaptedulce) [inline src/services/userDetailsApi.js]: remove this id from path

File: src/components/user/UserDetails.vue (hunk 5 above)
Comment (avolchivschi) [reply src/components/user/UserDetails.vue]: Username must be between 6 and 50 characters long right?

File: src/components/activity/ActivityList.vue [hunk 12]
```diff
@@ -155,7 +155,7 @@
             console.log (error)
           })
       },
-      
+
       setActivityId (id) {
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue]: rename as - redirectToActivityDatails

File: src/components/activity/VActivity.vue [hunk 13]
```diff
@@ -10,49 +10,49 @@
               <b-card-title class="col-md-8 mx-auto row">
                 {{ Activity.name }}
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: rename this object in - activity

File: src/components/auth/Login.vue [hunk 14]
```diff
@@ -25,7 +29,7 @@
               />
 
               <span
-                v-if="errors.has('username')"
+                v-if=" errors.has('username')"
```
Comment (nlaptedulce) [inline src/components/auth/Login.vue]: remove space after "

File: src/components/activity/CreateActivity.vue [hunk 15]
```diff
@@ -169,12 +169,7 @@
         formClone.technologies.push ({id: this.technologyId});
         formClone.types.push ({id: this.typeId});
         RegisterService.createActivity (formClone)
```
Comment (nlaptedulce) [inline src/components/activity/CreateActivity.vue]: rename formClone - activity

File: src/components/activity/CreateActivity.vue [hunk 16]
```diff
@@ -169,12 +169,7 @@
...
-
       },
```
Comment (nlaptedulce) [inline src/components/activity/CreateActivity.vue]: remove commas

File: src/components/auth/Login.vue [hunk 17]
```diff
@@ -47,7 +51,7 @@
               />
 
               <span
-                v-if="errors.has('password')"
+                v-if=" errors.has('password')"
```
Comment (nlaptedulce) [inline src/components/auth/Login.vue:54]: remove space after "

File: src/components/auth/Login.vue [hunk 18]
```diff
@@ -77,32 +81,35 @@
...
-        });
-    }
-  },
-};
+      loginError: '',
+    }),
+    methods: {
+      ...mapActions('account', ['login', 'logout']),
+      onSubmit() {
+        console.log('before login');
```
Comment (nlaptedulce) [inline src/components/auth/Login.vue]: remove console.log

File: src/components/auth/Login.vue [hunk 19]
```diff
@@ -77,32 +81,35 @@
...
+        LoginService.login(this.form)
+          .then(response => {
+            console.log('login data', response.data);
```
Comment (nlaptedulce) [inline src/components/auth/Login.vue]: remove console.log

File: src/components/auth/Login.vue [hunk 20]
```diff
@@ -77,32 +81,35 @@
...
+            const { token } = response.data;
+            window.localStorage.setItem('token', token);
+            this.login(token);
+            this.$router.push('/');
+          })
+          .catch(error => {
+            this.loginError = error;
+            this.logout();
```
Comment (nlaptedulce) [inline src/components/auth/Login.vue]: you don't need this here

File: src/components/auth/NotFound.vue [hunk 21]
```diff
@@ -0,0 +1,29 @@
+<template>
+  <div class="login-registration-background">
+    <b-container fluid>
+      <b-row align-h="center" class="mt-5">
+        <b-img :src="require('../../assets/images/error_bee.png')"/>
+        <div class="a">Oops!</div>
+      </b-row>
+      <div class="text-center">
```
Comment (nlaptedulce) [inline src/components/auth/NotFound.vue]: add a line before this div

File: src/components/auth/NotFound.vue [hunk 22]
```diff
@@ -0,0 +1,29 @@
+<template>
```
Comment (nlaptedulce) [inline src/components/auth/NotFound.vue]: add this file in folder views not components

File: src/components/auth/Registration.vue [hunk 23]
```diff
@@ -32,6 +32,26 @@
...
+              <b-form-input
+                id="email"
+                v-model="form.email"
+                v-validate.continues="'required|email'"
+                name="email"
+                type="email"
+                class="form-control"
+                :class="{ 'is-invalid': errors.has('email') }"
+              />
+              <span v-if="errors.has('email')" class="invalid-feedback">{{
```
Comment (nlaptedulce) [inline src/components/auth/Registration.vue]: make this part more readable

File: src/components/user/ChangePassword.vue [hunk 24]
```diff
@@ -1,39 +1,130 @@
...
         <b-form-input
+          id="confirmPassword"
+          v-model="form.confirm_password"
+          v-validate="'required|confirmed:password'"
+          name="confirm_password"
           type="password"
           class="form-control"
+          data-vv-as="password"
+          :class="{
+            'is-invalid': errors.has('confirm_password'),
```
Comment (nlaptedulce) [inline src/components/user/ChangePassword.vue]: write this in single line and remove coma

File: src/components/user/ChangePassword.vue [hunk 25]
```diff
@@ -1,39 +1,130 @@
...
+      ...mapActions('account',['login','logout']),
+      change() {
+        const body = {
+          ...this.form,
+          id:this.userId,
+        };
+        UserApi.changePassword(body)
+          .then(response => {
+            this.$router.push('/login');
+            console.log(response);
```
Comment (nlaptedulce) [inline src/components/user/ChangePassword.vue]: remove console.log

File: src/components/user/ChangePassword.vue [hunk 26]
```diff
@@ -1,39 +1,130 @@
...
+        old_password: '',
+        password: '',
+        confirm_password: '',
+      },
+    }),
+    computed:{
+      ...mapState('account',['user']),
+      userId(){
+
+        return this.user && this.user.id || UserApi.getUserDetails()
```
Comment (nlaptedulce) [inline src/components/user/ChangePassword.vue]: why you need here user if you have user.id

File: src/components/user/ChangePassword.vue [hunk 27]
```diff
@@ -1,39 +1,130 @@
...
+      },
+    },
+    methods: {
+      ...mapActions('account',['login','logout']),
+      change() {
+        const body = {
+          ...this.form,
+          id:this.userId,
```
Comment (nlaptedulce) [inline src/components/user/ChangePassword.vue]: space after :

File: src/components/user/ChangePassword.vue [hunk 28]
```diff
@@ -1,39 +1,130 @@
...
+      ...mapState('account',['user']),
+      userId(){
+
+        return this.user && this.user.id || UserApi.getUserDetails()
+      },
+    },
+    methods: {
+      ...mapActions('account',['login','logout']),
+      change() {
+        const body = {
```
Comment (nlaptedulce) [inline src/components/user/ChangePassword.vue]: rename in data

File: src/components/user/DeleteProfile.vue [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+        @click="onclick()"
+      > Delete Account
+      </b-btn>
+    </div>
+  </div>
+</template>
+
+<script>
+  import UserApi from '@/services/userDetailsApi';
+  import { mapActions,mapState } from 'vuex';
```
Comment (nlaptedulce) [inline src/components/user/DeleteProfile.vue]: space after comma

File: src/components/user/DeleteProfile.vue [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+
+  export default {
+    computed:{
+      ...mapState('account',['user']),
```
Comment (nlaptedulce) [inline src/components/user/DeleteProfile.vue]: space after comma

File: src/components/user/DeleteProfile.vue [hunk 31]
```diff
@@ -0,0 +1,45 @@
...
+      ...mapActions('account',['login','logout']),
+      onclick() {
+        const body = {
+        ...this.form,
+        id:this.userId,
+        };
+        UserApi.delete(body)
+          .then(response => {
+            this.$router.push('/login');
+            console.log(response);
```
Comment (nlaptedulce) [inline src/components/user/DeleteProfile.vue]: remove console.log

File: src/components/user/DeleteProfile.vue [hunk 32]
```diff
@@ -0,0 +1,45 @@
...
+      ...mapState('account',['user']),
+      userId(){
+        return this.user && this.user.id || UserApi.getUserDetails()
+      },
+    },
+    methods: {
+      ...mapActions('account',['login','logout']),
+      onclick() {
+        const body = {
+        ...this.form,
```
Comment (nlaptedulce) [inline src/components/user/DeleteProfile.vue]: why you need form?

File: src/components/user/EditProfile.vue [hunk 33]
```diff
@@ -0,0 +1,70 @@
...
+  import UserDetails from './UserDetails';
+  import ChangePassword from './ChangePassword';
+  import DeleteProfile from './DeleteProfile';
+  import UserApi from '@/services/userDetailsApi';
+
+  export default {
+    components: { UserDetails, ChangePassword, DeleteProfile },
+
+    methods: {
+      userInfo(id){
```
Comment (nlaptedulce) [inline src/components/user/EditProfile.vue]: why you need this method here?

File: src/components/user/UserDetails.vue [hunk 34]
```diff
@@ -1,66 +1,211 @@
...
+
+      UserApi.userInfo(this.userId).then((response) => {
+        this.form = response.data;
+      }).catch(error => {
+        console.log(error)
+      });
+
+      UserApi.getTechnologies(this.userId).then((response) => {
+        this.formTechnologies = response.data;
+        console.log(this.formTechnologies)
```
Comment (nlaptedulce) [inline src/components/user/UserDetails.vue]: remove console

File: src/components/user/UserDetails.vue [hunk 35]
```diff
@@ -1,66 +1,211 @@
...
+
+      }).catch(error => {
+        console.log(error);
+      });
+    },
+    methods: {
+      ...mapActions('account',['login','logout']),
+      edit() {
+        const body = {
```
Comment (nlaptedulce) [inline src/components/user/UserDetails.vue]: rename in data

File: src/components/user/UserDetails.vue [hunk 36]
```diff
@@ -1,66 +1,211 @@
...
+          ...this.form,
+          id:this.userId,
```
Comment (nlaptedulce) [inline src/components/user/UserDetails.vue]: add space after :

File: src/components/user/VProfile.vue [hunk 37]
```diff
@@ -1,53 +1,93 @@
...
+    }),
+    computed: {
+      seniorityList: () => ['JUNIOR', 'MIDDLE', 'SENIOR'],
+      ...mapState('account', ['user']),
+      userId() {
+        return UserApi.getUserDetails();
+      },
+    },
+    mounted() {
+      console.log(this);
```
Comment (nlaptedulce) [inline src/components/user/VProfile.vue]: remove console

File: src/services/Services.js [hunk 38]
```diff
@@ -1,8 +1,8 @@
-import TokenService from 'axios';
+import Axios from 'axios';
 import jwt_decode from 'jwt-decode';
 
-const TOKEN = localStorage.getItem ('token');
 const BASE_URL='http://api.pentabee.local/api/';
```
Comment (nlaptedulce) [inline src/services/Services.js]: use apiEndpoints

File: src/services/userDetailsApi.js [hunk 39]
```diff
@@ -0,0 +1,28 @@
+import Axios from 'axios';
+import { userEndpoints } from '@/constants/apiEndpoints';
+import jwt_decode from 'jwt-decode';
+
+export default {
+  getUserDetails() {
+    const token = localStorage.getItem('token');
+    const { id } = jwt_decode(token);
+    console.log(id);
```
Comment (nlaptedulce) [inline src/services/userDetailsApi.js]: remove console

File: src/store/index.js [hunk 40]
```diff
@@ -1,11 +1,13 @@
 import Vue from 'vue';
 import Vuex from 'vuex';
 import aside from './aside';
+import account from './user';
```
Comment (nlaptedulce) [inline src/store/index.js:4]: rename account in user

**Pull request merged**